### PR TITLE
Rename MakeStructTypeFromFields to MakeStructType

### PR DESCRIPTION
**Pull request by aboodman**

### CODE REVIEW

**Comment by arv** [comment]
jenkins: test this

File: go/types/type_cache.go [hunk 1]
```diff
@@ -403,20 +403,20 @@ func (fs *fieldSorter) Less(i, j int) bool {
...
 	}
-	fs := fieldSorter{names, types}
-	sort.Sort(&fs)
-	return MakeStructType(name, names, types)
+	return MakeStructTypeQuickly(name, fieldNames, fieldTypes)
 }
 
-func MakeStructType(name string, fieldNames []string, fieldTypes []*Type) *Type {
+func MakeStructTypeQuickly(name string, fieldNames []string, fieldTypes []*Type) *Type {
+	fs := fieldSorter{fieldNames, fieldTypes}
```
Comment (arv) [inline go/types/type_cache.go]: This is a change.

Let's make this function package private and not do the sorting.

File: samples/go/nomsfs/nomsfs.go [hunk 2]
```diff
@@ -100,13 +100,25 @@ type mount func(fs pathfs.FileSystem)
...
-	inodeType = types.MakeStructType("Inode", []string{"attr", "contents"}, []*types.Type{
-		types.MakeStructType("Attr", []string{"ctime", "gid", "mode", "mtime", "uid", "xattr"}, []*types.Type{types.NumberType, types.NumberType, types.NumberType, types.NumberType, types.NumberType, types.MakeMapType(types.StringType, types.BlobType)}),
-		types.MakeUnionType(types.MakeStructType("Directory", []string{"entries"}, []*types.Type{
-			types.MakeMapType(types.StringType, types.MakeCycleType(1))}),
-			types.MakeStructType("File", []string{"data"}, []*types.Type{types.MakeRefType(types.BlobType)}),
-			types.MakeStructType("Symlink", []string{"targetPath"}, []*types.Type{types.StringType}),
-		),
+	inodeType = types.MakeStructType("Inode", types.FieldMap{
+		"attr": types.MakeStructType("Attr", types.FieldMap{
+			"ctime": types.NumberType,
```
Comment (arv) [inline samples/go/nomsfs/nomsfs.go]: This is now readable 👏

**Comment by arv** [comment]
jenkins: test this please

File: go/types/type_cache.go (hunk 1 above)
Comment (aboodman) [inline go/types/type_cache.go]: I don't understand this comment.

**Comment by arv** [comment]
Test failed but the log is now gone.

**Comment by arv** [comment]
BTY

**Comment by arv** [comment]
Closing this. I will tackle it as part of #2327

**Pull request closed without merging**